scripts/bqeth_subscription.py

```diff
@@ -0,0 +1,267 @@
+    tx_hash = context.subscription_contract.functions.payForSubscription(encryptor_slots).transact(
+        {"from": context.account.address}
+    )
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+
+@cli.command()
+@click.option("--extra-slots", default=1, help="Number of additional encryptor slots to pay for.")
+@setup_context
+def pay_slots(context, extra_slots):
+    """Pay for additional encryptor slots."""
+    click.echo(f"Paying for {extra_slots} new encryptor slots.")
+    tx_hash = context.subscription_contract.functions.payForEncryptorSlots(extra_slots).transact(
+        {"from": context.account.address}
+    )
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+
+@cli.command()
+@click.option("--num-nodes", default=2, help="Number of nodes to use for the ritual.")
+@click.option("--duration", help="Duration in seconds for the ritual.", type=int)
+@click.option("--random-seed", help="Random seed integer for sampling.")
+@click.option("--num-rituals", default=1, help="Number of rituals to initiate.", type=int)
+@setup_context
+def initiate_ritual(context, num_nodes, duration, random_seed, num_rituals):
+    """Initiate a ritual."""
+    if NUCYPHER_DOMAIN == "mainnet":
+        porter_request = f"https://porter.nucypher.community/bucket_sampling?quantity={num_nodes}&random_seed={random_seed}"
+    else:
+        porter_request = f"{PORTER_ENDPOINT}/get_ursulas?quantity={num_nodes}"
+
+    # start_of_subscription = context.subscription_contract.functions.startOfSubscription().call()
+    # duration = (
+    #     context.subscription_contract.functions.subscriptionPeriodDuration().call()
+    #     + context.subscription_contract.functions.yellowPeriodDuration().call()
+    #     + context.subscription_contract.functions.redPeriodDuration().call()
+    # )
+    # if start_of_subscription > 0:
+    #     click.echo(
+    #         "Subscription has already started. Subtracting the elapsed time from the duration."
+    #     )
+    #     now = context.w3.eth.get_block("latest")["timestamp"]
+    #     elapsed = now - start_of_subscription + 100
+    #     duration -= elapsed
+
+    click.echo(f"Account address that will sign the transaction: {context.account.address}")
+    click.echo(f"Initiating ritual with {num_nodes} providers for {duration} seconds.")
+    click.echo(f"Endpoint used: {PROVIDER_URL}")
+
+    for i in range(num_rituals):
+        click.echo(f"Initiating ritual {i + 1} of {num_rituals}.")
+        nodes = sorted(
+            [
+                u["checksum_address"]
+                for u in requests.get(porter_request).json()[
+                    "result"
+                ]["ursulas"]
+            ], key=lambda x: x.lower()
+        )
+
+        tx_hash = context.coordinator_contract.functions.initiateRitual(
+            context.subscription_contract.address,
+            nodes,
+            context.account.address, # TODO: parametrize this
```

Comment:
This is the authority address of the ritual.  For production use-cases this will not be the same as the initiator so we need to parameterize this.